src/Icons/config/services.php

```diff
@@ -53,6 +53,7 @@
                 service('logger')->ignoreOnInvalid(),
             ])
             ->tag('twig.runtime')
+            ->tag('.ux.twig_component.twig_renderer', ['key' => 'ux:icon'])
```

Comment:
can you please double confirm this works with attribute `key`?
should the $needsIndexes argument of TaggedIterator be set to true?
sorry I don't remember, evethought I might have written the logic years ago :)
also, doesn the tag really need to be "private"? it should be fine to add the the list to me, that's the whole point of the DI container of SYmfony: out of the box extensibility

Reply:
> also, doesn the tag really need to be "private"?

My idea was to keep this functionality internal until a time we add a proper interface/document its usage. It's not exactly specified in Symfony's BC policy but I wanted to ensure no one uses this tag until then for BC purposes.

Reply:
the thing is that changing the name of a tag is hard from a BC pow, so I'd recommend using a proper name (and no need for any interface nor doc for now at least, ppl are responsible for wiring their things properly)

Reply:
Makes sense!